src/main/java/com/onlinebookstore/onlinebookstore/controller/OrderController.java

```diff
@@ -0,0 +1,81 @@
+package com.onlinebookstore.onlinebookstore.controller;
+
+import com.onlinebookstore.onlinebookstore.dto.order.OrderItemResponseDto;
+import com.onlinebookstore.onlinebookstore.dto.order.OrderRequestDto;
+import com.onlinebookstore.onlinebookstore.dto.order.OrderResponseDto;
+import com.onlinebookstore.onlinebookstore.dto.order.OrderUpdateStatusDto;
+import com.onlinebookstore.onlinebookstore.model.User;
+import com.onlinebookstore.onlinebookstore.service.interfaces.OrderService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/orders")
+@Tag(name = "Order endpoints management", description = "Endpoints to manage shopping cart")
+public class OrderController {
+    private final OrderService orderService;
+
+    @GetMapping
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Get all orders", description = " User can view all order "
+            + "their details and status")
+    public Set<OrderResponseDto> getOrderHistory(
+            @AuthenticationPrincipal User user) {
+        return orderService.getOrderHistory(user);
+    }
+
+    @GetMapping("/{orderId}/items")
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Get a all items in order", description = " User can view "
+            + "all items in order")
+    public Set<OrderItemResponseDto> getOrderItems(
+            @PathVariable Long orderId) {
+        return orderService.getOrderItems(orderId);

```

Comment:
Please make sure that one user can`t access order data of different user (check all endpoints)